code/modules/black_market/illegalradio.dm

```diff
@@ -0,0 +1,370 @@
+
+	dat += "<br><HR><font size=3><A href='byond://?src=\ref[src];open_main=1'>Return</a></font>"
+	dat = jointext(dat,"") 
+	return dat	
+
+		
+		
+/obj/item/device/illegalradio/proc/open_screen(var/screen)
+	opened_screen = screen
+	interact(usr)
+		
+/obj/item/device/illegalradio/proc/generate_new_local_listing(atom/target)
+	new_listing = new /datum/black_market_player_item()
+	new_listing.item = target
+	new_listing.selected_name = "[target]"
+	new_listing.seller_radio = src
+	new_listing.seller = usr
+	new_listing.selected_price = minimum_price
+	open_screen(SELLING)
+		
+/obj/item/device/illegalradio/afterattack(atom/movable/A as mob|obj, mob/user as mob)
+	if(istype(A, /obj/item/weapon/spacecash) && A.Adjacent(user))
+		var/obj/item/weapon/spacecash/cash = A
+		money_stored += cash.get_total()
+		qdel(cash)
+		visible_message("<span class='info'>[usr] inserts a credit chip into [src].</span>")
+		interact(usr)
+	else if((istype(A, /obj) || istype(A, /mob)) && A.Adjacent(user) && !scanning)
+		visible_message("\The [src] beeps: <span class='warning'>Scanning item to sell...</span>")
+		scanning = 1
+		if(do_after(user, A, scan_time))
+			if(user.z != STATION_Z)
+				visible_message("\The [src] beeps: <span class='warning'>Error: you must be near your space station. The connection is poor in deep space.</span>")
+			else if(A.anchored)
+				visible_message("\The [src] beeps: <span class='warning'>Error: that item is anchored. Can't teleport the entire station with it.</span>")
+			else
+				visible_message("\The [src] beeps: <span class='warning'>Item verified. Please confirm your listing.</span>")
+				generate_new_local_listing(A)
+		scanning = 0		
+		
+/obj/item/device/illegalradio/MouseDropTo(var/atom/movable/target, var/mob/user)
+	afterattack(target, user)
+	
+
+/obj/item/device/illegalradio/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/weapon/spacecash))
+		var/obj/item/weapon/spacecash/C = W
+		insert_cash(C, user)
+		interact(user)
+	
+/obj/item/device/illegalradio/emag_act(mob/user)
+	visible_message("<span class='warning'>[usr] swipes a card through [src], and it explodes!</warning>")
+	explosion(user, -1, 0, 2)
+	to_chat(user, "<span class='notice'>You hear a faint laughter in your head.<span>")
+	qdel(src)
+
+/obj/item/device/illegalradio/proc/insert_cash(var/obj/item/weapon/spacecash/C, mob/user)
+	visible_message("<span class='info'>[usr] inserts a credit chip into [src].</span>")
+	money_stored += C.get_total()
+	qdel(C)
+
+/obj/item/device/illegalradio/proc/dispense_change()
+	if(money_stored > 0)
+		dispense_cash(money_stored,get_turf(src))
+		money_stored = 0
+	interact(usr)
+	
+/obj/item/device/illegalradio/proc/open_html(var/dat_input)
+	var/dat = "<body link='yellow' alink='white' bgcolor='#331461'><font color='white'>"
+	dat += dat_input
+	dat += "</body></font>"
+	usr << browse(dat, "window=hidden")
+	onclose(usr, "hidden")	
```

Comment:
Try to find a way to not use `usr` here.

Reply:
This gets called exclusively from interact(user), so you can just add a user arg to the proc and pass it